src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -274,4 +274,14 @@ public function addResource(ResourceInterface $resource)
     {
         $this->resources[] = $resource;
     }
+
+    public function serialize()
+    {
+        return serialize($this->routes);
+    }
+
+    public function unserialize($serialized)
+    {
+        $this->routes = unserialize($serialized);
```

Comment:
serializing and unserializing a collection is loosing the resources. This means that generating the UrlMatcher from a collection loaded from the cache will not generate the proper meta file